### PR TITLE
5118 add service + endpoint to create a response requisition

### PR DESCRIPTION
**Pull request by roxy-dao**
Part of #5118

# 👩🏻‍💻 What does this PR do?
Service and endpoint to create response requisitions (normal).

## PS ADDING LINES COMING IN NEXT PR

# 🧪 Testing

- [ ] Use GraphQL to call new endpoint

# 📃 Documentation

- [ ] **Part of an epic**: documentation will be completed for the feature as a whole
- [x] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [ ] **These areas should be updated or checked**: 
  1.
  2.

### CODE REVIEW

File: server/graphql/requisition/src/mutations/response_requisition/insert.rs [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+#[graphql(name = "InsertResponseRequisitionInput")]
+pub struct InsertInput {
+    pub id: String,
+    pub other_party_id: String,
+    pub max_months_of_stock: f64,
+    pub min_months_of_stock: f64,
+}
+
+#[derive(Interface)]
+#[graphql(name = "InsertResponseRequisitionErrorInterface")]
```
Comment (fergie-nz) [inline server/graphql/requisition/src/mutations/response_requisition/insert.rs:27]: I wonder do we rename graphql fields throughout the code? I find the renaming slightly confusing to trace types, but maybe that is just me?

Comment (roxy-dao) [reply server/graphql/requisition/src/mutations/response_requisition/insert.rs:27]: Yes, we use this for the Object name throughout the codebase.

Comment (Chris-Petty) [reply server/graphql/requisition/src/mutations/response_requisition/insert.rs:27]: (Not requesting any change!)

>Yes, we use this for the Object name throughout the codebase.

You mean the frontend code base? I think Ferg's point is that when trying to trace backend implementation it's not always intuitive to search for `InsertErrorInterface` which is generically named within this code context, but in a code search things become harder to find (and when working you have to be constantly aware of directory structure of the file you're editing to apply context to the variable name - bit of a cognitive load when trying to jump around different files.

image

Trade off though - once you are contextualised the short names are cozy and it favours copy pasting all this boiler plate.

That said not everywhere has the different names:

https://github.com/msupply-foundation/open-msupply/blob/7a31855c50d62f04c22ac8014bc10bf3abdef6e5/server/graphql/requisition/src/mutations/request_requisition/insert_program.rs#L17-L19

**Review by Chris-Petty — APPROVED** [review]
Looks good to me!

File: server/graphql/requisition/src/mutations/response_requisition/insert.rs (hunk 1 above)
Comment (roxy-dao) [reply server/graphql/requisition/src/mutations/response_requisition/insert.rs:27]: Ah that's true. I usually go through by folder name then action to find them 🤔

**Pull request merged**